bm_serial.c

```diff
@@ -292,36 +306,50 @@ bm_serial_error_e bm_serial_set_rtc(bm_serial_time_t *time) {
   \param[in] *fw_info
   \param[in] num_nodes
   \param[in] *node_id_list
+  \param[in] config_map_size
+  \param[in] *cbor_config_map
   \return BM_SERIAL_OK if sent, nonzero otherwise
 */
-bm_serial_error_e bm_serial_send_network_info(uint32_t network_crc32, bm_common_config_crc_t *config_crc, bm_common_fw_version_t *fw_info, uint16_t num_nodes, uint64_t* node_id_list) {
+bm_serial_error_e bm_serial_send_network_info(
+    uint32_t network_crc32, bm_common_config_crc_t *config_crc,
+    bm_common_fw_version_t *fw_info, uint16_t num_nodes, uint64_t *node_id_list,
+    uint16_t config_map_size, uint8_t *cbor_config_map) {
   bm_serial_error_e rval = BM_SERIAL_OK;
   do {
 
-    if (!config_crc || !fw_info || !node_id_list || num_nodes == 0) {
+    if (!config_crc || !fw_info || !node_id_list || num_nodes == 0 ||
+        !cbor_config_map) {
       rval = BM_SERIAL_MISC_ERR;
       break;
     }
 
-    uint16_t message_len = sizeof(bm_serial_packet_t) + sizeof(bm_common_network_info_t) + (sizeof(uint64_t) * num_nodes);
+    uint16_t message_len = sizeof(bm_serial_packet_t) +
+                           sizeof(bm_common_network_info_t) +
+                           (sizeof(uint64_t) * num_nodes) + config_map_size;
 
-    bm_serial_packet_t *packet = _bm_serial_get_packet(BM_SERIAL_NETWORK_INFO, 0, message_len);
+    bm_serial_packet_t *packet =
+        _bm_serial_get_packet(BM_SERIAL_NETWORK_INFO, 0, message_len);
 
     if (!packet) {
       rval = BM_SERIAL_OUT_OF_MEMORY;
       break;
     }
 
-    bm_common_network_info_t *network_info = (bm_common_network_info_t *)packet->payload;
+    bm_common_network_info_t *network_info =
+        (bm_common_network_info_t *)packet->payload;
     network_info->network_crc32 = network_crc32;
-    memcpy(&network_info->config_crc, config_crc, sizeof(bm_common_config_crc_t));
+    memcpy(&network_info->config_crc, config_crc,
+           sizeof(bm_common_config_crc_t));
     memcpy(&network_info->fw_info, fw_info, sizeof(bm_common_fw_version_t));
     network_info->node_list.num_nodes = num_nodes;
-    memcpy(&network_info->node_list.list, node_id_list, sizeof(uint64_t)*num_nodes);
+    memcpy(&network_info->node_list.list, node_id_list,
+           sizeof(uint64_t) * num_nodes);
+    network_info->config_map.map_size_bytes = config_map_size;
+    memcpy(&network_info->config_map.map, cbor_config_map, config_map_size);
```

Comment:
Since the node_list.list is a zero length array, will line 347 put the `network_info->config_map.map_size_bytes` in the right spot or do we need to do something like: `memcpy(&network_info->nodelist.list[num_nodes], config_map_size, sizeof(config_map_size));`?

Reply:
Oh hmm good point, probably not?  Your suggestion is probably safer, I'll change it to that.

Reply:
Actually I likely have to change the order in bm_common_messages. 🤦 The double zero-length thing is not going to work well.

Reply:
yeah - since these are pretty custom messages we could make the network_info have:
```
uint8_t num_nodes;
uint8_t map_len'
uint8_t node_list_and_map[0];
```

Reply:
this is how it is done for other messages, and thus we can access the map by doing `node_list_and_map[num_nodes];`

Reply:
NVM you did exactly that already! 😂